### PR TITLE
Validate config and fail fast 

### LINKED ISSUE #222: Validate the configuration and fail-fast
e.g. required fields for deploy vs provision, sane values for CPU, memory

### PR DESCRIPTION
**Pull request by philipstaffordwood**
closes #222

### CODE REVIEW

File: pkg/platform/platform.go [hunk 1]
```diff
@@ -805,3 +817,52 @@ func (platform *Platform) DefaultNamespaceAnnotations() map[string]string {
...
+	err := validateCluster(platform)
+	if err != nil {
+		return err
+	}
+
+	if platform.Master.CPUs < 2 {
+		return fmt.Errorf(ErrInsufficientVms)
+	}
+
+	if platform.Calico.Version == "" {
```
Comment (moshloop) [inline pkg/platform/platform.go:859]: Either a calico or nsx version needs to be specified

File: pkg/platform/platform.go [hunk 2]
```diff
@@ -805,3 +817,52 @@ func (platform *Platform) DefaultNamespaceAnnotations() map[string]string {
...
+		return fmt.Errorf(ErrNoCalicoVersion)
+	}
+
+	if platform.Consul == "" {
```
Comment (moshloop) [inline pkg/platform/platform.go:863]: With recent changes this is no longer true - You can use either NSX only, DNS only, Consul only or Consul + DNS

File: pkg/platform/platform.go (hunk 1 above)
Comment (philipstaffordwood) [reply pkg/platform/platform.go]: see if https://github.com/flanksource/karina/pull/365/commits/7702f5754edb6089ccf05e1516ec9bfdc5851eb4 resolves this satisfactorily

**Comment by moshloop** [comment]
Merged manually

**Pull request closed without merging**